### ISSUE TITLE
 [SUGGESTIONS] Future Customizations and Wiki Content

### ISSUE THREAD

**Issue by Hypnootize** [Hypnootize/m0rehud#39]
Post here any suggestion about future customizations and new content for the wiki pages, feel free to also give feedback about the suggested and already present customizations.

### - Sniper Scope Charge Positioning by [czarchasm](https://github.com/Hypnootize/m0rehud/issues/35)
Currently covered in the wiki, might add it as customization in the future

### - Low Health Warning by [czarchasm](https://github.com/Hypnootize/m0rehud/issues/37)
Will mention this in the wiki soon, adding a dedicated customization in the future

### - Control Point Progress Bar by [qkeitoe](https://github.com/Hypnootize/m0rehud/issues/38)
Will be added to the customizations in the future

### - Damage Indicators Size by Hypnotize
Will be added to the customizations in the future, could be worth covering how to manually resize them in the wiki

### - Direct Hitmarker Customizability by Hypnotize
Could be worth to making it a thing in the future. Definitely no need to run the full set of crosshairs there as well, pick the most fitting ones and add the option to run them as hitmarkers from the customizations panel.

### - Deeper Color Customization From The Customizations Panel by Hypnotize
Kinda unlikely for this to become reality, but it would be technically possible to add direct customization for each one of the color aliases inside `colors_scheme.res`. Users would be able to pick from a set of preset colors and apply them to each element.

&nbsp; 
&nbsp; 

### - Bring Back Overtime Label
I feel like it's the only one worth adding back, but if they fit nicely might be ok to keep the others as well

### - Add High Killstreak to WinPanel.res
I find it super useless but people like it, I'll think of a good way to implement this eventually

**Comment by qkeitoe** [comment]
### HUD THEMES

I think the fact that this option affects not only the health color, but also the color of full uber, main menu, class selection menu and so on, is a bit odd. In particular, recently, when trying to change the health style (so that the numbers themselves become colored, not their shadows), I realized that this can only be achieved manually. It seems to me (and since I've seen a similar setup among a few streamers) that such a desire is relatively common.

**Comment by Hypnootize** [comment]
That's what the `Deeper Color Customization` would aim to do, as even before m0re 7.0 the themes would change all colors and if you wanted to you could manually fine tune them.

The main problem behind all of the theme and color setup is that you can't edit the animation files efficiently by using base logging, if you were able to simply toggle the animation style then there wound't be any need to have double theme buttons to color either the shadow or the number for example, it could have been a simple on/off button that worked with selected color, sadly that's not the case so I'm left finding workarounds.

What I could do is implement a color selection system for each one of the aliases inside `colors_scheme.res`, or at least the main ones, so you can easily select which color you want for each element (Health Buff/Low and were to apply it to for example) from a list of preset colors, kinda like it works for the themes right now.
This should be doable even tho it might end up being a little messy code wise but that doesn't concern me too much. It would be something similar to how the TF2 Editor manages the colors for my HUDs right now.

**Comment by qkeitoe** [comment]
Are you going to reimplement flashing ammo this way as well?

**Comment by Hypnootize** [comment]
It's a possibility, it would mean having the flash animation running by default tho, which isn't a problem by itself but it might make things a bit more confusing when applying the colors.

**Comment by czarchasm00** [comment]
I think the ability to remove menu buttons would be really solid. I find that quite a few of them I just simply don't care for.

What should be removable (importance high to low):
-Store (Scam)
-Version (There is an okay reason for this but it's so niche it's not worth typing out)
-Workshop (Good for people who create assets for TF2, but this is a tiny audience)
-Minmode (This should be just a straight up customization instead of a hud button; i struggle to imagine a scenario where someone actively changes minmode in-game)
-DemoUI (I use this all the time but I know for more casual users they will see no purpose in this button)
-Create (Not useful as long as you know the map command and map prefixes; i feel many players will also just never have a purpose for this)
-Toggle Netgraph (This is alright but I don't use netgraph often enough to justify it having its own button so prefer customization option)

**Comment by qkeitoe** [comment]
I don't think many people care about extra buttons, since any button doesn't mess with anything and sometimes is extremely useful. For example, the store button is useful after major updates (for previewing new hats/taunts).

However, it seems like such customization option is not too tricky to implement (just change `""pin_to_sibling_corner"` from `"PIN_BOTTOMLEFT"` to `"PIN_TOPLEFT"`).

And I strongly feel that for the version button such a feature should not be added (the HUD name and the link to the source are inherent parts of the main menu).

**Comment by r3dnasa** [comment]
a centered mode would be nice, having the health and ammo closer together towards the center of the screen to make it easier to see at a glance

**Comment by Hypnootize** [comment]
> I think the ability to remove menu buttons would be really solid. I find that quite a few of them I just simply don't care for.

I personally don't find this necessary, a wiki page featuring a guide on how to add and remove buttons should be good to have tho. 
I'm planning to move the minmode button inside the HUD customizations panel, as for the other buttons imo they add value without any obstruction or annoyances to users that might never use them.

> a centered mode would be nice, having the health and ammo closer together towards the center of the screen to make it easier to see at a glance

This is not a bad idea, ideally I would have liked to have a couple of presets that let you move Health, Ammo and Uber closer or further away to the center, you are very limited on what you can do simply because there is way too much stuff in the center of the screen if you count all possible element that could be enabled at the same time. 

I'll leave this for the future for now but I think I'll eventually come up with something.

#

I have been mostly busy with other stuff lately but I have started adding the possibility to edit colors independently today

![HUDCOLORS](https://github.com/Hypnootize/m0rehud/assets/19785551/60052ef0-4452-4635-8ee3-e5f3fcd25e24)

It's only missing the files to make it work but that's the easy part. I'd also like to add all the suggestion above for the next release but we'll see.

**Comment by qkeitoe** [comment]
I think it would be useful to have a wiki page about how to hide medals in scoreboard (perhaps this could even be put in a dedicated customization), especially since with dxlevel 8, it's almost impossible to distinguish medals by rank and level.

**Comment by Hypnootize** [comment]
I agree on the scoreboard medals, having a wiki section on how to enable and disable them would be good. I got asked how to re-enable them for the fosters scoreboard twice already so I kinda had plans to write something about it already.

For the main menu idk, I don't find it super annoying and I mostly play on DX8, I'd like to exclude menu edits from the customizations for now.

**Comment by czarchasm00** [comment]
If the user launches with -console in their launch options, and then opens up the Friends menu, it is partially blocked by the console. Having a customization to put it on the left side would be pretty neat. (IDK if you can change the position of where the console initially opens, so if you can please let me know)

https://i.imgur.com/fiEQdU5.png

**Comment by qkeitoe** [comment]
Since this panel is rarely needed (few times per game session), it seems to me that it is not a big deal to move/close the console at these moments.

However, thinking about possible locations for the friends list, I remembered that some older version had it on the top left (where the party chat pops up). I'm not sure, but it seems that such a location would make some sense in the current version, since the friend list would be right under the button that opens it (rather than at almost the opposite end of the screen).

**Comment by Hypnootize** [comment]
As far as I'm aware you can change the initial console location on game launch, Jofre managed to make it work iirc, but it's simply so tricky that it isn't worth it in the end.
As for the positioning, both left side and top left under the party memeber have their issues. In the left side (basically mirroring the current position) it will cause problems for 4:3, 5:4 and so on, while for the top left positioning it will overlap with the party notification and party invites making it impossible to invite multiple friends, I did used to have it there but then moved it where it is now for that reason. The current location seems to be the safest one.

**Comment by qkeitoe** [comment]
Oh, I see.

Then what do you think about moving both the button and the list to the top right, or placing the button vertically at the center right (where the list is located, hiding/showing the button when the list opens/closes)? Not that I'm in any way bothered by the current placement (as I said, this panel and button are only used few times per game session), it's just that conceptually having the button and the panel it enables so far apart seems a bit odd.

**Comment by qkeitoe** [comment]
Or maybe even something like that
![image](https://github.com/Hypnootize/m0rehud/assets/66339491/421d7af4-68ca-4371-ae2c-204d804204fa)

**Comment by qkeitoe** [comment]
I've been playing around a bit with possible variants of placing the friends list, and here's an idea I came up with. I'm not quite sure if it's user friendly or if it causes any bugs, but this looks cool to me. I realize that on other resolutions the panel wide has to be changed, but it doesn't seem to be a problem.
![Screenshot (28)](https://github.com/Hypnootize/m0rehud/assets/66339491/ecd5d076-4a27-432b-9417-ef954f387fda)

**Comment by Hypnootize** [comment]
It honestly does looks good but it doesn't feel like the most intuitive thing to go look inside the matchmaking panel for the friends list, plus the resolution aspect but you mentioned that.
It feels like looking for a solution to a problem that isn't really there or isn't too major. I don't find having to resize a little bit the console on game launch too annoying if in the other hand you get a well functioning, intuitive and bug free list, but that's just my point of view.

**Comment by Hypnootize** [comment]
The last comming on the [dev branch](https://github.com/Hypnootize/m0rehud/tree/m0re-7.1-dev) features most of the suggested customizations as well as some things I had already plans to include
```
- Ammo Flash
- Crosshair Flash
- Hitmarkers
- Uber Counter
- Transparent Damage Numbers
- Damage Indicators Customization
- Scope Charge Customization
- Health Warning Percentage Customization
- Control Point Progress Bar ON/OFF Customization
- Killstreak Notice ON/OFF Customization
```

Feel free to [download and test the version](https://github.com/Hypnootize/m0rehud/archive/refs/heads/m0re-7.1-dev.zip)

**Comment by qkeitoe** [comment]
Potential idea for a wiki page: How to disable the stats panel and the duel leaderboard on the loading screen

**Comment by Hypnootize** [comment]
Sounds good, I can add it to the additions list. Now that the new version is mostly complete I'll start adding some newer content to the wiki.

**Comment by qkeitoe** [comment]
I think it would be great to have additional scoreboard styles, something like hexhud has (short and full stats styles) or similar to ahud / broeselhud one.

**Comment by Hypnootize** [comment]
Could be interesting yeah. I'm not sure when I'll be looking to implement more as of now but I can definitely keep the idea for the future.

**Comment by billw1** [comment]
idk if anyone else would actually use it but I like the idea of [hurt](https://i.imgur.com/ND2XZ9E.png)/[buffed](https://i.imgur.com/tk4ebg2.png) vignettes like the Overwatch hud has done. they could be done in a much more subtle/generic way that fits tf2 a bit more

**Comment by Sizzler48** [comment]
Player with highest killstreak added to round summary screen?

**Comment by Skincanvas** [comment]
An option to make the match/round timer huge, or even centered next to the crosshair, would be great. Its only really relevant for competitive but a lot of plays are based on the timer and being able to see it easier would be very nice.

**Comment by Skincanvas** [comment]
Also not sure where I should put this, but I made a script to automate the path fixing for Linux which seems to work well. Maybe something like this could be included in master so people don't have to wait for a branch release.
[fixlinux.txt](https://github.com/user-attachments/files/17510318/fixlinux.txt)
Rename m0rehud to m0rehud-linux, put the script in the m0rehud-linux folder, then run it. I assume it will work on future releases.

**Comment by Sizzler48** [comment]
> Also not sure where I should put this, but I made a script to automate the path fixing for Linux which seems to work well. Maybe something like this could be included in master so people don't have to wait for a branch release. [fixlinux.txt](https://github.com/user-attachments/files/17510318/fixlinux.txt) Rename m0rehud to m0rehud-linux, put the script in the m0rehud-linux folder, then run it. I assume it will work on future releases.

Just tried it out as I daily drive Linux: works super well! Thank you!

**Comment by LukasFromPoland** [comment]
maybe adding box target id like you have in m0re black?

![Image](https://github.com/user-attachments/assets/0b4c8289-1174-4544-9303-1f5b89af9503)

**Comment by RhyliaNoski** [comment]
Any way to add outline to Health/Ammo in the in-game hud? maybe also in the main menu font? been trying to figure out how to add this on my own by having trouble with it.

**Comment by LukasFromPoland** [comment]
@RhyliaNoski yes you can, add to you're font parameter outline like this and it should work

![Image](https://github.com/user-attachments/assets/f3d2ee06-8965-48a8-8e1a-5076ca9c2ebd)

**Comment by RhyliaNoski** [comment]
> [@RhyliaNoski](https://github.com/RhyliaNoski) yes you can, add to you're font parameter outline like this and it should work
> 
> ![Image](https://github.com/user-attachments/assets/f3d2ee06-8965-48a8-8e1a-5076ca9c2ebd)

this works, thank you. But is there a command to make the outline bigger? i tried playing with the dropshadows but it doesn't seem to work well.

**Comment by LukasFromPoland** [comment]
@RhyliaNoski I don't think so, but to be honest, I don't know

**Comment by Bear-B34R** [comment]
-Could you add a customization option for boxes around the health/ammo?
-Also could you have it so health values are consistent across everything?
For example, whenever I have no health cross on player it doesn't carry over to target
or if I set it so the number itself is coloured, things like the person who killed me isn't the same way but the default shadow.
Just general consistency across the hud.
